### DISCUSSION TITLE
Generic File Manager

### DISCUSSION THREAD

**Discussion by ekristen** [renovatebot/renovate#11677] [Suggest an Idea]
What do you think about a generic file manager? 

I have this use case, I know a few others do as well, where there's a file, either generated as an asset on a GitHub release or generated to some directory path in the repository itself that you want/need to copy into another project but doesn't fall under any existing package management capability.

I think this could apply to a single file or a directory of files. Really anytime you need to share something that is generated but doesn't align itself to a traditional package manager.

### Use Case 1

Cloudposse has a terraform-null-label module that generates a context.tf that's in an export directory, they then copy it to all their other modules as context.tf when they tag a new version, it seems to me that renovate could easily handle this situation.

### Use Case 2

I have several projects that are sources for a CRD instance for kubernetes, each one has their own defaults for the YAML, right now I attache the generated file as an asset to the tagged github release, and then I have to manually copy the entire yaml into the GitOps repositories, I could probably script this though.

When it's just a docker image, or a version update, using the regex feature works well enough, but when there's substantial amount of changes to the underlying YAML that doesn't hold up as those other changes need to be pulled over.

**Comment by rarkins** [comment]
Yes, I like the idea and have mentioned it before although not sure if there was ever an issue for it.

I think it would be helpful if you can convert your use cases to public reproduction repos, ideally with a hand-crafted PR showing the type of update you'd need (including proposed PR title).

It's not fully obvious about how to fit this model of changed file/files into our existing workflow either. e.g. what "updateType" do we classify it as, do we need new fields passed to `datasource`, etc?

Another challenge for us is how to do this efficiently. It would not be great if we are downloading every file on every run "just in case" they've changed. Ideally there'd be a way of caching at least the ETag or similar so that we can avoid that.

**Reply by HonkingGoose** [reply]
> Yes, I like the idea and have mentioned it before although not sure if there was ever an issue for it.

We have an issue discussing a "generic datasource", is that what you're thinking of?

- https://github.com/renovatebot/renovate/issues/10622

**Reply by rarkins** [reply]
No, that one's for supporting the traditional releases approach

**Reply by HonkingGoose** [reply]
Oh okay, sorry for getting it wrong! 😄

**Reply by rarkins** [reply]
No, that one's for a traditional releases-based endpoint

**Reply by rarkins** [reply]
I found the issue: #4759
